### ISSUE TITLE
Test case: zfs_written_property_001_pos

### ISSUE THREAD

**Issue by FransUrbo** [zfsonlinux/zfs#2441]
Way to reproduce:

```
#!/bin/bash

get_prop() {
        zfs get -pH -o value $1 $2
}

within_percent() {
        a=$1
        b=$1
        percent=$3

        # Set $a or $b to $2 such that a >= b
        [[ '1' = $(echo "if ($2 > $a) 1" | bc) ]] && a=$2 || b=$2

        # Prevent division by 0
        [[ $a =~ [1-9] ]] || return 1

        p=$(echo "scale=2; $b * 100 / $a" | bc)
        [[ '1' = $(echo "scale=2; if ($p >= $percent) 1" | bc) ]] && return 0

        return 1
}

rm -f /var/tmp/zfs_test-1
truncate -s 25000m /var/tmp/zfs_test-1
zpool create test /var/tmp/zfs_test-1
zfs create -p test/test1

total=0; blocks=50 ; mb_block=`expr 1024 \* 1024`
for i in {1..3}; do
        zfs snapshot test/test1@snap$i

        dd if=/dev/urandom of=/test/test1/testfile.$i bs=1M count=$blocks \
            > /dev/null 2>&1 ; sync

        written=$(get_prop written test/test1)
        expected_written=`expr $blocks \* $mb_block`

        echo "($i) Written '$written', expected to write: '$expected_written'"
        if within_percent $written $expected_written 99.5; then
                echo "    SUCCESS: Expected written value $expected_written is within %99.5 of actual written $written"
        else
                echo "    FAIL: Unexpected written value $written (expected '$expected_written')"
        fi

        blocks=`expr $blocks + 50`
        echo
done
```

Results in:

```
# /tmp/testme.sh
(1) Written '52489216', expected to write: '52428800'
    SUCCESS: Expected written value 52428800 is within %99.5 of actual written 52489216

(2) Written '56686592', expected to write: '104857600'
    FAIL: Unexpected written value 56686592 (expected '104857600')

(3) Written '110197760', expected to write: '157286400'
    FAIL: Unexpected written value 110197760 (expected '157286400')
```

**Comment by behlendorf** [comment]
What's going on here is that under Linux `sync` doesn't force a txg sync.  It does make sure the data is safe on disk and that may be in the ZIL.  Since the accounting takes place at txg sync time the written value may be slightly incorrect.  This is expected behavior.  One possible solution would be to add a `zfs sync` command to force the txg sync which can be used by the test suite.

**Comment by behlendorf** [comment]
Resolved by 95401cb6f7aa9ec3dd2d61335d2dcc9459c2661c.
